pymc3/distributions/logprob.py

```diff
@@ -141,8 +130,88 @@ def logpt(
         A scaling term to apply to the generated log-likelihood graph.
     transformed
         Apply transforms.
-    cdf
-        Return the log cumulative distribution.
+
+    """
+    if not isinstance(rv_values, Mapping):
+        rv_values = (
+            {var: at.as_tensor_variable(rv_values).astype(var.type)}
+            if rv_values is not None
+            else {}
+        )
+
+    tmp_rv_values = {}
+    transform_map = {}
+    for node in io_toposort(graph_inputs((var,)), (var,)):
+        if isinstance(node.op, RandomVariable):
+            curr_var = node.out
+            rv_value_var = getattr(
+                curr_var.tag, "observations", getattr(curr_var.tag, "value_var", curr_var)
+            )
+            rv_value = rv_values.get(curr_var, rv_value_var)
+            tmp_rv_values[curr_var] = rv_value
```

Comment:
```suggestion
            tmp_rvs_to_values[curr_var] = rv_value
            tmp_values_to_rvs[rv_value] = curr_var
```
Then you don't need the list of keys thing in the loop below